martini_sour/src/analyze.py

```diff
@@ -0,0 +1,175 @@
+# Copyright 2021 University of Groningen
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import MDAnalysis as mda
+from tqdm import tqdm
+import matplotlib.pyplot as plt
+from scipy.optimize import curve_fit
+from .titrate import get_ph_values
+
+
+def setup_dirs(ph_range, sim_dir, traj_file, tpr_file):
+    """
+    Sets up directory paths based on whether multidir options have been provided or not.
+    """
+
+    if ph_range and sim_dir:
+        ph_values = get_ph_values(ph_range)
+        traj_paths = [f'{ph}/{sim_dir}/{traj_file}' for ph in ph_values]
+        tpr_paths = [f'{ph}/{sim_dir}/{tpr_file}' for ph in ph_values]
+    else:
+        traj_paths = [traj_file]
+        tpr_paths = [tpr_file]
+
+    return traj_paths, tpr_paths, np.array(ph_values, dtype=float)
+
+
+def save_to_file(out_file, ph_values, props, sel_commands):
+    array = np.concatenate((ph_values[np.newaxis], props.T), axis=0)
+
+    fmt = ['%6.2f']+['%10.5f']*(len(array)-1)
```

Comment:
Does np.array only allows the old formatting way? Using f formats or a format string would be better.